pkg/client/client_test.go

```diff
@@ -42,10 +42,23 @@ func TestNewClientOK(t *testing.T) {
 func TestNewClientFail(t *testing.T) {
 	// when
 	cl, err := client.NewClient("cool-token", "https://fail-cluster.com")
-
+	require.NoError(t, err)
+	assert.NotNil(t, cl)
 	// then
+	testObj := &toolchainv1alpha1.UserSignup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "john-doe",
+			Namespace: "default",
+		},
+	}
+	// no specific reason to check if object is namespaced, ANY request to the actual api would trigger error indicating incorrect configuration of client
+	_, err = cl.IsObjectNamespaced(testObj)
 	require.Error(t, err)
-	assert.Nil(t, cl)
+	// actual error is "failed to get restmapping: failed to get server groups: Get \"https://fail-cluster.com/api?timeout=1m0s\": dial tcp: lookup fail-cluster.com: no such host"
+	require.ErrorContains(t, err, "dial tcp: lookup fail-cluster.com")
+	// for ci the error message is dial tcp: lookup fail-cluster.com on 127.0.0.53:53: no such host, could go the regex way or string submatching with wildcards
+	// having two separate checks for different substrings is an easy fix
+	require.ErrorContains(t, err, "no such host")
```

Comment:
I did consider only checking for "no such host", but I do see value in checking for the host name as well - which we are setting explicitly in the test above.